### PR TITLE
0.12 PCF Shadow Filtering

### PR DESCRIPTION
**Pull request by cart**
Part of #754

### CODE REVIEW

File: content/news/2023-10-21-bevy-0.12/index.md [hunk 1]
```diff
@@ -15,6 +15,45 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+
+![no pcf high resolution](no_pcf_high.png)
+
+Looking better! However this still isn't a perfect solution. Large shadowmaps aren't feasible on all hardware. And even if they are, you can still encounter this issue if you place an object in the wrong place, or point your light in the wrong direction. You can use Bevy's [Cascaded Shadow Maps](/news/bevy-0-10/#cascaded-shadow-maps) (which are enabled by default) to cover a larger area, with higher detail close to the camera and less detail farther away. However even under these conditions, you will still probably encounter these aliasing issues.
+
+**Bevy 0.12** introduces **PCF Shadow Filtering**, which is a popular technique that takes multiple samples from the shadow map and compares with an interpolated mesh surface depth projected into the frame of reference of the light. It then calculates the percentage of samples in the depth buffer that are closer to the light than the mesh surface. In short, this creates a "blur" effect that improves shadow quality, which is especially evident when a given shadow doesn't have enough "shadow map detail".
```
Comment (rparrett) [inline content/news/2023-10-21-bevy-0.12/index.md:34]:
```suggestion
**Bevy 0.12** introduces **PCF Shadow Filtering**, which is a popular technique that takes multiple samples from the shadow map and compares with an interpolated mesh surface depth-projected into the frame of reference of the light. It then calculates the percentage of samples in the depth buffer that are closer to the light than the mesh surface. In short, this creates a "blur" effect that improves shadow quality, which is especially evident when a given shadow doesn't have enough "shadow map detail".
```

Is this how I'm supposed to read this series of words? :)

Comment (rparrett) [inline content/news/2023-10-21-bevy-0.12/index.md:34]: Should we define this acronym here (or elsewhere)?

File: content/news/2023-10-21-bevy-0.12/index.md [hunk 2]
```diff
@@ -15,6 +15,45 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+
+![no pcf](no_pcf.png)
+
+Those "jagged lines" in the shadow are the result of the shadow map being "too small" to accurately represent the shadow from this perspective. The shadow map above is stored in a 512x512 texture.
+
+One "solution" is to bump up the resolution. Here is what it looks like with a 4096x4096 shadow map.
+
+![no pcf high resolution](no_pcf_high.png)
+
+Looking better! However this still isn't a perfect solution. Large shadowmaps aren't feasible on all hardware. And even if they are, you can still encounter this issue if you place an object in the wrong place, or point your light in the wrong direction. You can use Bevy's [Cascaded Shadow Maps](/news/bevy-0-10/#cascaded-shadow-maps) (which are enabled by default) to cover a larger area, with higher detail close to the camera and less detail farther away. However even under these conditions, you will still probably encounter these aliasing issues.
```
Comment (JMS55) [inline content/news/2023-10-21-bevy-0.12/index.md:32]: Large shadow maps take a lot of memory and more importantly rendering time. I think saying hardware capabilities is misleading, it's more about performance.

**Comment by JMS55** [comment]
512x512 also looks a bit bad, even with PCF. Maybe you want to double it.

**Comment by cart** [comment]
> 512x512 also looks a bit bad, even with PCF. Maybe you want to double it.

Hmmm yeah that was kind of the point. Show an "egregious" case. I was also worried that 1024 would benefit too much from "linear filtering" when sampling the scaled image in web browsers, making the effect less prominent.

Imo 1024 isn't _significantly_ better. And the difference is less stark (left is 512 right is 1024)
image

**Comment by cart** [comment]
Resolved comments. I also explicitly called out that we don't use 512x512 shadowmaps by default / it was chosen to illustrate the point.

**Pull request merged**